cyfronet-fid/marketplace
# feat:[#3046] Add research products to the Projects

Add research products pinnable to the projects
Add models, controllers, and routes

How to test PR:
go to the new research_product_path
`/research_products/new`
with resource_id and resource_type params,
eg.
```
/research_products/new?resource_id=50|dedup_wf_001%3A%3A00e804d55fff4d449f2b105245992ae0&resource_type=software
/research_products/new?resource_id=50|06cdd3ff4700%3A%3A0d0916f71e57fc252109f47376a83194&resource_type=publication
```

then you can see an "add to the project" form
In the details, you can delete research_product
Closes https://github.com/cyfronet-fid/marketplace/issues/3046


Figma: https://www.figma.com/file/l6nViF0XSdT98of1VjJV9f/EOSC-Pin-the-Project?type=design&node-id=5-1639&mode=design&t=yjCOLQ4IXrROnODt-0
